### ISSUE TITLE
ui: Vault and Consul ui_url not available without ACL token

### ISSUE THREAD

**Issue by shoeffner** [hashicorp/nomad#14461]
### Nomad version
Output from `nomad version`
`Nomad v1.3.4 (3bed4df9e498e747278cd7f4c509884812751ed6+CHANGES)`

### Issue

Since v1.3.3, more specifically since https://github.com/hashicorp/nomad/blame/61fd9436ab046bbb8c31bee9de23229cfd2dce33/ui/app/services/system.js#L39-L40, the UI does no longer show the Vault and Consul links for users who did not provide an ACL token in the top navigation bar.

This is because the request to `/v1/agent/self` is now guarded and requires an ACL token to be set, even if the anonymous policy has the correct [agent:read](https://www.nomadproject.io/api-docs/agent#query-self) rights.

### Reproduction steps
Create `nomad_server.hcl` and the anonymous policy below:

*nomad_server.hcl*
```hcl
ui {
  enabled = true
  vault {
    ui_url = "https://vault-address.example.org/"
  }
}
```

*anonymous.policy.hcl*:
```hcl
namespace "*" {
  policy = "deny"
}

agent {
  policy = "write"
}

operator {
  policy = "write"
}

quota {
  policy = "write"
}

node {
  policy = "write"
}

host_volume "*" {
  policy = "write"
}
```

Then run nomad and bootstrap it:
```console
$ ./bin/nomad agent -dev -acl-enabled -config=nomad_server.hcl
# in a different terminal:
$ ./bin/nomad acl bootstrap
...
Secret ID    = abcde
...
$ export NOMAD_TOKEN=abcde
$ ./bin/nomad acl policy apply -description "Anonymous policy" anonymous anonymous.policy.hcl
```

After that, open the UI: http://localhost:4646/ui
In case you have a token set (e.g., the management token created above), you can, after reloading, see the Vault link:
![image](https://user-images.githubusercontent.com/1836815/188395838-260ee51b-57e8-4f4e-996d-74eb6f7be36e.png)
Otherwise, its space stays vacant.

#### Expected Result

The Vault UI link (and Consul UI link) should be accessible without a token.

#### Actual Result

The Vault UI link (and Consul UI link) are not accessible without a valid token.

**Comment by ednxzu** [comment]
The problem has been addressed in #14359 and I believe release [v1.3.5](https://github.com/hashicorp/nomad/releases/tag/v1.3.5) fixed it, tho I did not update yet.

**Comment by shoeffner** [comment]
Thanks, you are right, it's fixed with #14359 .
